aten/src/ATen/InferSize.h

```diff
@@ -41,12 +46,24 @@ inline std::vector<int64_t> infer_size(IntArrayRef shape, int64_t numel) {
                "value and is ambiguous");
       res[*infer_dim] = numel / newsize;
     }
-    return res;
+    return;
   }
 
   std::ostringstream ss;
   ss << "shape '" << shape << "' is invalid for input of size " << numel;
   throw std::runtime_error(ss.str());
 }
 
+inline std::vector<int64_t> infer_size(IntArrayRef shape, int64_t numel) {
+  auto res = shape.vec();
+  infer_size_impl(res, shape, numel);
```

Comment:
nit: putting the out parameter last means you won't have to shift arguments around if somehow infer_size and/or infer_size_impl don't get inlined